### PR TITLE
sql: add CREATE TABLE LIKE

### LINKED ISSUE #30840: sql: Provide CREATE TABLE ... LIKE syntax
The functionality desired would be something similar to MySQL's CREATE TABLE ... LIKE functionality. Which is described here: https://dev.mysql.com/doc/refman/5.5/en/create-table-like.html

In Cockroach today the closet you get is a CREATE TABLE AS SELECT * ... LIMIT 0. However, this approach doesn't automatically create any indices that the original table might have had.

### PR DESCRIPTION
**Pull request by jordanlewis**
Closes #30840.

CREATE TABLE LIKE allows the user to create a table based on the schema
of a different table. Different combinations of INCLUDING specifiers
allow the user to customize which bits of the different table should be
copied into the new table.

By default, CREATE TABLE a (LIKE b) will include all of the columns of b
into a, including their nullability, and nothing else.

This commit adds the specifiers:

- INCLUDING CONSTRAINTS adds all CHECK constraints from the source
- INCLUDING DEFAULTS adds all DEFAULT column expressions from the source
- INCLUDING GENERATED adds all computed column expressions from the source
- INCLUDING INDEXES adds all indexes from the source table
- INCLUDING ALL includes all of the above specifiers.

Column families, interleavings, partitioning and foreign key constraints
cannot be preserved from the old table and will have to be recreated
manually in the new table if the user wishes.

In addition, users can exclude specifiers using the EXCLUDING keyword.
This is useful to include ALL and then remove things that aren't wanted.
The most recent INCLUDING/EXCLUDING for a given specifier wins.

LIKE specifiers can also be mixed with ordinary CREATE TABLE specifiers.

For example:

```
CREATE TABLE a (a INT PRIMARY KEY, b INT NOT NULL DEFAULT 3 CHECK (b > 0), INDEX(b))

CREATE TABLE dest (LIKE a INCLUDING ALL EXCLUDING CONSTRAINTS, c INT, INDEX(b,c))
```

The dest table will get all of the indexes and defaults and so on from
a, but not its check constraints, because `EXCLUDING CONSTRAINTS` was
specified after `INCLUDING ALL`.

Also includes `sql: slightly improve hash sharded index creation`

- improve type checking of bucket count for sharded indexes
- return an error if a user happens to name a column the same thing as
  one of the hidden columns for hash sharded indexes.

Release note (sql change): small UX improvements to hash sharded index
creation

Release note (sql change): add CREATE TABLE LIKE specifiers. See commit
message for more detailed examples and documentation.

### CODE REVIEW

**Comment by jordanlewis** [comment]
@ajwerner if you could take a look at the little fixes to hash-sharded indexes in the first commit, that would be great!

File: pkg/cmd/docgen/diagrams.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func runParse(
 		return nil, errors.Wrap(err, "inline")
 	}
 	b, err := g.ExtractProduction(topStmt, descend, nosplit, match, exclude)
-	b = bytes.Replace(b, []byte("IDENT"), []byte("identifier"), -1)
+	b = bytes.Replace(b, []byte("'IDENT'"), []byte("'identifier'"), -1)
```
Comment (rohany) [inline pkg/cmd/docgen/diagrams.go:296]: What does this change fix?

File: pkg/sql/create_table.go [hunk 2]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
 			createStmt = &newCreateStmt
 		}
 	}
+	// First, process "CREATE TABLE LIKE" statements by moving the definitions from
+	// the referenced table into the AST being processed.
+	var newDefs tree.TableDefs
+	for i, def := range n.Defs {
```
Comment (rohany) [inline pkg/sql/create_table.go:1704]: are we allowed to have multiple likes in one create table statement?

File: pkg/sql/create_table.go [hunk 3]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
...
+				newDefs = append(newDefs, def)
+			}
+			continue
+		}
+		// We're definitely going to be editing n.Defs now, so make a copy of it.
+		if newDefs == nil {
+			newDefs = make(tree.TableDefs, 0, len(n.Defs))
+			newDefs = append(newDefs, n.Defs[:i]...)
+		}
+		td, err := ResolveMutableExistingObject(params.ctx, params.p, &d.Name, true /*required*/, ResolveRequireTableDesc)
```
Comment (rohany) [inline pkg/sql/create_table.go:1717]: I think you want to use the planners table collection here

File: pkg/sql/create_table.go [hunk 4]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
...
+		if err != nil {
+			return ret, err
+		}
+		opts := tree.LikeTableOpt(0)
+		// Process ons / offs.
+		for _, opt := range d.Options {
+			if opt.Excluded {
+				opts &^= opt.Opt
```
Comment (rohany) [inline pkg/sql/create_table.go:1725]: What does this operator do lol

File: pkg/sql/create_table.go [hunk 5]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
...
+		defs := make(tree.TableDefs, 0)
+		// Add all columns. Columns are always added.
+		for i := range td.Columns {
+			c := &td.Columns[i]
+			if c.Hidden {
+				// Hidden columns automatically get added by the system; we don't need
+				// to add them ourselves here.
+				continue
+			}
+			def := tree.ColumnTableDef{
```
Comment (rohany) [inline pkg/sql/create_table.go:1740]: I think this should be pulled out into a function taking a column descriptor and spitting out a column table def, given some options

File: pkg/sql/create_table.go [hunk 6]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
...
+				def.Expr, err = parser.ParseExpr(c.Expr)
+				if err != nil {
+					return ret, err
+				}
+				defs = append(defs, &def)
+			}
+		}
+		if opts.Has(tree.LikeTableOptIndexes) {
+			for _, idx := range td.AllNonDropIndexes() {
+				indexDef := tree.IndexTableDef{
```
Comment (rohany) [inline pkg/sql/create_table.go:1783]: same comment as above for these indexes

Comment (rohany) [reply pkg/sql/create_table.go:1783]: We also do this operation a bunch in some virtual tables, can possibly unify these

File: pkg/sql/create_table.go [hunk 7]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
...
+					if isPK && td.IsPrimaryIndexDefaultRowID() {
+						continue
+					}
+
+					def = &tree.UniqueConstraintTableDef{
+						IndexTableDef: indexDef,
+						PrimaryKey:    isPK,
+					}
+				}
+				fmt.Println("Adding index def", tree.AsString(&indexDef))
```
Comment (rohany) [inline pkg/sql/create_table.go:1821]: print

File: pkg/sql/create_table.go [hunk 8]
```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
 			createStmt = &newCreateStmt
 		}
 	}
+	// First, process "CREATE TABLE LIKE" statements by moving the definitions from
+	// the referenced table into the AST being processed.
+	var newDefs tree.TableDefs
```
Comment (rohany) [inline pkg/sql/create_table.go:1703]: Do we want like table to use the same family definitions too?

File: pkg/sql/logictest/testdata/logic_test/create_table [hunk 9]
```diff
@@ -110,3 +110,208 @@ WHERE feature_name IN (
 ----
 sql.schema.inverted_index
 sql.schema.hash_sharded_index
+
+subtest like_table
+
+# Test the CREATE TABLE LIKE functionality.
+
+statement ok
+CREATE TABLE like_table (
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/create_table:119]: specify the families here or the column family randomizer will rek u

File: pkg/sql/logictest/testdata/logic_test/create_table [hunk 10]
```diff
@@ -110,3 +110,208 @@ WHERE feature_name IN (
...
+                  b STRING NOT NULL,
+                  c DECIMAL NULL,
+                  h INT8 NULL,
+                  j JSONB NULL,
+                  FAMILY "primary" (a, b, c, h, j, rowid),
+                  CONSTRAINT check_a CHECK (a > 3)
+)
+
+statement ok
+CREATE TABLE like_indexes (LIKE like_table INCLUDING INDEXES)
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/create_table:163]: is there any wierd behavior with having a unique constraint + an index and then saying like including indexes and constraints? Basically looking at the case where the constraints and indexes overlap

File: pkg/sql/logictest/testdata/logic_test/create_table [hunk 11]
```diff
@@ -110,3 +110,208 @@ WHERE feature_name IN (
...
+              j JSONB NULL,
+              CONSTRAINT "primary" PRIMARY KEY (a ASC, b ASC),
+              UNIQUE INDEX foo (b DESC, c ASC),
+              INDEX like_table_c_idx (c ASC) STORING (j),
+              INVERTED INDEX like_table_j_idx (j),
+              FAMILY "primary" (a, b, c, h, j)
+)
+
+statement ok
+CREATE TABLE like_generated (LIKE like_table INCLUDING GENERATED)
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/create_table:182]: a comment in this case for what you are expecting to see please

File: pkg/sql/logictest/testdata/logic_test/create_table [hunk 12]
```diff
@@ -110,3 +110,208 @@ WHERE feature_name IN (
...
+          CONSTRAINT "primary" PRIMARY KEY (a ASC, b ASC),
+          UNIQUE INDEX foo (b DESC, c ASC),
+          INDEX like_table_c_idx (c ASC) STORING (j),
+          INVERTED INDEX like_table_j_idx (j),
+          FAMILY "primary" (a, b, c, h, j),
+          CONSTRAINT check_a CHECK (a > 3)
+)
+
+statement ok
+CREATE TABLE like_mixed (LIKE like_table INCLUDING ALL EXCLUDING GENERATED EXCLUDING CONSTRAINTS INCLUDING GENERATED)
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/create_table:232]: what does this even mean

File: pkg/sql/parser/sql.y [hunk 13]
```diff
@@ -4522,7 +4531,41 @@ table_elem:
...
+    $$.val = append($1.likeTableOptionList(), opt)
+  }
+| /* EMPTY */
+  {
+    $$.val = []tree.LikeTableOption(nil)
+  }
+
+like_table_option:
+  COMMENTS			{ return unimplemented(sqllex, "like table in/excluding comments") }
+| CONSTRAINTS		{ $$.val = tree.LikeTableOption{Opt: tree.LikeTableOptConstraints} }
```
Comment (rohany) [inline pkg/sql/parser/sql.y:4560]: add the unimplemented tests in the parse_test for these cases

**Review by rohany — CHANGES_REQUESTED** [review]

**Review by ajwerner — COMMENTED** [review]
Hash sharded changes :lgtm: 

Added a suggestion, take it or leave it. 

> Reviewed 2 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-:-M4EcegC2pT75ZTaOS_U:bi39urt)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner and @jordanlewis)

___
*[pkg/sql/sqlbase/table.go, line 241 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M4EcLgoAGspjiGZdKUP:-M4EcLgp-mtwO6kivdCh:boangms) ([raw file](https://github.com/cockroachdb/cockroach/blob/0c6d317d86d35a4c9706e71fda18c36240c511e4/pkg/sql/sqlbase/table.go#L241)):*
> ```Go
> 	}
> 	buckets := tree.MustBeDInt(d)
> 	if buckets <= 0 {
> ```

should we complain about `1`? A hash sharded index with one bucket is sort of pointless.

**Review by jordanlewis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47017)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner and @rohany)

___
*[pkg/cmd/docgen/diagrams.go, line 296 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCS-r2-296:-M4CgBuzBqP9S22k6OgT:bhiiua0) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/cmd/docgen/diagrams.go#L296)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

What does this change fix?
</blockquote></details>

It's dumb - the IDENTITY keyword was the first keyword in our grammar that continained IDENT but wasn't just IDENT on its own. So the regex was turning IDENTITY into identifierITY :P 
___
*[pkg/sql/create_table.go, line 1703 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1703:-M4CgMq_DK86Fr59OpQg:bdru3bl) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1703)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

Do we want like table to use the same family definitions too?
</blockquote></details>

I was thinking that that could be a future improvement, as it is fairly annoying. (we could add INCLUDING FAMILIES). but maybe i should just add now... thoughts?
___
*[pkg/sql/create_table.go, line 1704 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1704:-M4CgVUO8o3jVYXI7tnt:b-hwuwmg) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1704)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

are we allowed to have multiple likes in one create table statement?
</blockquote></details>

Yes. And it's tested later on.
___
*[pkg/sql/create_table.go, line 1717 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1717:-M4HslZNBaLN7OJcvMSd:b-a94f9y) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1717)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

I think you want to use the planners table collection here
</blockquote></details>

Believe it or not, this thing eventually does use that? Everyone seems to go through this interface, so I will too. I've changed it to use the method version which seems to be identical...
___
*[pkg/sql/create_table.go, line 1725 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1725:-M4CgYB129GyPJ41NdJZ:bj5ettw) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1725)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

What does this operator do lol
</blockquote></details>

It's the "remove bitmask" operator. And not.
___
*[pkg/sql/create_table.go, line 1740 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1740:-M4HtE4pCaOjKUab-ZoR:bwg1fgd) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1740)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

I think this should be pulled out into a function taking a column descriptor and spitting out a column table def, given some options
</blockquote></details>

But this is the only place that is going to use the function. Normally I'd agree... I've moved the whole LikeTableDef processing thing into its own function at least.
___
*[pkg/sql/create_table.go, line 1783 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1783:-M4HsxsVEYaVbIwPHU5J:b-rn5sdk) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1783)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

We also do this operation a bunch in some virtual tables, can possibly unify these
</blockquote></details>

They're slightly different though. I think it's probably easier to just do it in both places.
___
*[pkg/sql/create_table.go, line 1821 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1821:-M4CgeQ-Cjh95iVB4ip1:b-kk8sdq) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1821)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

print
</blockquote></details>

Oops. Done.
___
*[pkg/sql/logictest/testdata/logic_test/create_table, line 119 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCT-r2-119:-M4Expmg9DxNYxZlCGp6:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/logictest/testdata/logic_test/create_table#L119)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

specify the families here or the column family randomizer will rek u
</blockquote></details>

Done.
___
*[pkg/sql/logictest/testdata/logic_test/create_table, line 163 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCT-r2-163:-M4Ex_IC3ayjellJK81M:b-6t7l7x) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/logictest/testdata/logic_test/create_table#L163)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

is there any wierd behavior with having a unique constraint + an index and then saying like including indexes and constraints? Basically looking at the case where the constraints and indexes overlap
</blockquote></details>

constraints only includes check constraints, according to the docs. (I've included that detail in the commit message too).
___
*[pkg/sql/logictest/testdata/logic_test/create_table, line 182 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCT-r2-182:-M4ExYLd8tJG5HgUp_fs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/logictest/testdata/logic_test/create_table#L182)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

a comment in this case for what you are expecting to see please
</blockquote></details>

Done.
___
*[pkg/sql/logictest/testdata/logic_test/create_table, line 232 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCT-r2-232:-M4ExJ8aCTXsbRZy8qjI:bin54um) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/logictest/testdata/logic_test/create_table#L232)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

what does this even mean
</blockquote></details>

It's in the commit message - it take the most recent value for each of the attributes.
___
*[pkg/sql/parser/sql.y, line 4560 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCV-r2-4560:-M4ChCy7803ur5em4w4u:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/parser/sql.y#L4560)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

add the unimplemented tests in the parse_test for these cases
</blockquote></details>

Done.
___
*[pkg/sql/sqlbase/table.go, line 241 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M4EcLgoAGspjiGZdKUP:-M4HtmTH0LHzXKGUVM6x:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0c6d317d86d35a4c9706e71fda18c36240c511e4/pkg/sql/sqlbase/table.go#L241)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

should we complain about `1`? A hash sharded index with one bucket is sort of pointless. 
</blockquote></details>

Done.

**Review by rohany — APPROVED** [review]
LGTM with that unimplemented nit

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-:-M4K1QY3E6s1SzWjTz_G:b-oq9ef8)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner, @jordanlewis, and @rohany)

___
*[pkg/cmd/docgen/diagrams.go, line 296 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbGko9_g1pppFCS-r2-296:-M4K0CisE14O__BL0zB6:bn65x3p) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/cmd/docgen/diagrams.go#L296)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

It's dumb - the IDENTITY keyword was the first keyword in our grammar that continained IDENT but wasn't just IDENT on its own. So the regex was turning IDENTITY into identifierITY :P 
</blockquote></details>

hmmmm :yacc:
___
*[pkg/sql/create_table.go, line 1703 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1703:-M4K10H_BXYmMYS5QwqA:b3tv99u) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1703)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

I was thinking that that could be a future improvement, as it is fairly annoying. (we could add INCLUDING FAMILIES). but maybe i should just add now... thoughts?
</blockquote></details>

Its up to you whether you should implement it in this PR, but I'd ad least add the syntax with an unimplemented error
___
*[pkg/sql/create_table.go, line 1717 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1717:-M4K0TEtD_yi_k7Tmvzv:by8rieb) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1717)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

Believe it or not, this thing eventually does use that? Everyone seems to go through this interface, so I will too. I've changed it to use the method version which seems to be identical...
</blockquote></details>

Oh ok, I thought they were different things
___
*[pkg/sql/create_table.go, line 1725 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1725:-M4K0l9S7sYmqgLVLf7P:b2nhoia) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1725)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

It's the "remove bitmask" operator. And not.
</blockquote></details>

TIL
___
*[pkg/sql/create_table.go, line 1740 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1740:-M4K0s-g1-t9z3Dtri0O:bwuel0p) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1740)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

But this is the only place that is going to use the function. Normally I'd agree... I've moved the whole LikeTableDef processing thing into its own function at least.
</blockquote></details>

OK, sg

**Review by jordanlewis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47017)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner, @jordanlewis, and @rohany)

___
*[pkg/sql/create_table.go, line 1703 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/47017#-M413fbFWeE7U-WMPpjX-r2-1703:-M4WsyU96mlqMsN6ry7t:b6rtsnl) ([raw file](https://github.com/cockroachdb/cockroach/blob/dc7edab9f9f95b666a6c86e9d72117f16cf510f3/pkg/sql/create_table.go#L1703)):*
<details><summary><i>Previously, rohany (Rohan Yadav) wrote…</i></summary><blockquote>

Its up to you whether you should implement it in this PR, but I'd ad least add the syntax with an unimplemented error
</blockquote></details>

Eh - but nobody will know to try doing it. It's not a real command yet. I'll add it later, assuming people even want the feature which so far guessing they won't.

**Comment by jordanlewis** [comment]
TFTR

bors r+

**Comment by jordanlewis** [comment]
Roachtest timeout.

bors r+

**Pull request merged**